client/homebrew/editor/snippetbar/snippetbar.jsx

```diff
@@ -59,13 +63,17 @@ const Snippetbar = createClass({
 
 	componentDidUpdate : async function(prevProps) {
 		if(prevProps.renderer != this.props.renderer || prevProps.theme != this.props.theme || prevProps.snippetBundle != this.props.snippetBundle) {
-			const snippets = this.compileSnippets();
 			this.setState({
-				snippets : snippets
+				snippets : this.compileSnippets()
 			});
-		}
-	},
+		};
 
+		if(historyExists(this.props.brew) != this.state.historyExists){
+			this.setState({
+				historyExists : !this.state.historyExists
+			});
+		};
```

Comment:
React is smart enough to not re-render if the new state value is the same as the old value. We don't have to do our own check if the value is different.
```suggestion
		this.setState({
			historyExists : historyExists(this.props.brew)
		});
```